File: objc/Project7/Project7/Photo.m
//
//  Photo.m
//  Project7
//
//  Created by Jinwoo Kim on 12/30/20.
//

#import "Photo.h"

@implementation Photo

- (void)viewDidLoad {
    [super viewDidLoad];
    self.selectedBorderThickness = 3;
    self.view.wantsLayer = YES;
    self.view.layer.borderColor = NSColor.blueColor.CGColor;
}

- (void)setSelected:(BOOL)selected {
    [super setSelected:selected];
    if (selected) {
        self.view.layer.borderWidth = self.selectedBorderThickness;
    } else {
        self.view.layer.borderWidth = 0;
    }
}

- (void)setHighlightState:(NSCollectionViewItemHighlightState)highlightState {
    [super setHighlightState:highlightState];
    if (highlightState == NSCollectionViewItemHighlightForSelection) {
        self.view.layer.borderWidth = self.selectedBorderThickness;
    } else {
        if (!self.isSelected) {
            self.view.layer.borderWidth = 0;
        }
    }
}
@end

File: swift/Project7/Project7/Photo.swift
//
//  Photo.swift
//  Project7
//
//  Created by Jinwoo Kim on 12/29/20.
//

import Cocoa

class Photo: NSCollectionViewItem {
    
    let selectedBorderThickness: CGFloat = 3
    
    override func viewDidLoad() {
        super.viewDidLoad()
        view.wantsLayer = true
        view.layer?.borderColor = NSColor.blue.cgColor
    }
    
    override var isSelected: Bool {
        didSet {
            if isSelected {
                view.layer?.borderWidth = selectedBorderThickness
            } else {
                view.layer?.borderWidth = 0
            }
        }
    }
    
    override var highlightState: NSCollectionViewItem.HighlightState {
        didSet {
            if highlightState == .forSelection {
                view.layer?.borderWidth = selectedBorderThickness
            } else {
                if !isSelected {
                    view.layer?.borderWidth = 0
                }
            }
        }
    }
}
